todo-app/src/components/ListItem.js

```diff
@@ -1,64 +1,62 @@
 import React from "react";
 import PropTypes from "prop-types";
+import { inject, observer } from "mobx-react";
 import UpdateTodo from "./UpdateTodo";
 
 
-const ListItem = (({ todoItems, todo }) => {
-
-    const {
-        toggleCheck,
-        toggleEdit,
-        deleteTodo
-    } = todoItems;
-
-    const {
-        id,
-        description,
-        deadline,
-        done,
-        editMode
-    } = todo;
-
-    const date = deadline.split("-").join("/");
-    const crossOut = done ? "crossOut" : null;
-
-    return (
-        <section>
-            {
-                editMode
-                ?
-                <UpdateTodo todo={todo}
-                    todoItems={todoItems}
-                />
-                :
-                <div className="list-item">
-                    <input type="checkbox"
-                        checked={done}
-                        onChange={() => toggleCheck(id)}
-                        className="checkbox"
-                    />
-                    <label className={crossOut}>
-                        {description},
-                        {new Date(date).toDateString()}
-                    </label>
-                    <button className="edit-btn"
-                        onClick={() => toggleEdit(id)}>
-                        Edit
-                    </button>
-                    <button className="delete-btn"
-                        title="delete"
-                        onClick={() => deleteTodo(id)}>
-                        X
-                    </button>
-                </div>
-            }
-        </section>
-    );
-})
-
+@inject("todoItems")
+@observer    
+export default class ListItem extends React.Component {
+    
+    render() {
+        const {
+            toggleCheck,
+            editID,
+            editTodo,
+            deleteTodo,
+        } = this.props.todoItems;
+
+        const {
+            id,
+            description,
+            deadline,
+            done
+        } = this.props.todo;
+
+        const date = deadline.split("-").join("/");
+        const crossOut = done ? "crossOut" : null;
+
+        return (
+            <section>
+                { editID === id
+                    ?
+                    <UpdateTodo todo={this.props.todo} />
```

Comment:
Remember that you can spread objects into props too, to avoid having to do `this.props.todo.id`. You can do:

    <UpdateTodo {...this.props.todo} />

Just an option!